www/docs/errors.mdx

```diff
@@ -53,51 +49,59 @@ async function runExample() {
 }
 ```
 
-When treating a task as a promise, if the task becomes halted, the promise is
-rejected with a special halt error:
+However, when an operation is halted, it will never produce a value,
+nor will it ever raise an error. And, because it will produce neither
+a positive nor negative outcome, it is an error to `await` the result
+of a halted operation.
+
+When  this happens, the promise is rejected with a special halt error:
 
 ``` typescript
-import { run } from 'effection';
+import { run, suspend } from 'effection';
 
 async function runExample() {
   try {
-    let task = run();
-    task.halt();
+    let task = run(supsend);
+    await task.halt();
     await task;
   } catch(err) {
     console.log("got error", err.message) // => "got error halted"
   }
 }
 ```
```

Comment:
I was following along up until this point. A lot is going on in this example, and something unexpected can be distracting. 

1. `run(suspend)` was confusing because it's unclear what this is doing. `run(function*(){})` might be less confusing because at least it's visible.
2. You're calling `task.halt()` in the same try/catch block as creating the task and awaiting for task result. It's a lot to process; it's unclear what the error will be because the code is not executable. If there were an inline runner, then it would make the result a bit clearer. The comment is not apparent. 

It might be helpful to break up the code somehow or show where the error will occur.

Reply:
@taras Have a look now. I think I made it a lot more explicit about what's going on.

Reply:
This is much better. Does `await task.halt()` throw an error? If it does, can we add this comment `await task.halt(); // 🔥`

Reply:
It actually doesn't, and this is an important point. `task.halt()` is an operation that only fails if there is a failure in the teardown of the task.

This is why I added the text right after the code.

> Notice how it is not an error to await the task.halt() operation, only to await the outcome of the operation which has been halted.

```ts
try {
  yield* suspend();
} finally {
  throw new Error('this halt operation will fail');
}
```

In other words "halting" is an operation that can succeed or fail in its own right. However, once a task is halted, then it will never return a value ever, and so an attempt to wait for its value is an error.

Reply:
> task.halt() is an operation that only fails if there is a failure in the teardown of the task.

Is more clear than `Notice how it is not an error to await the task.halt() operation, only to await the outcome of the operation which has been halted.`

```ts
  try {
    // halt it. It will no longer result in "hello world"
    await task.halt();

    await task; // 💥 throws halted error
  } catch(err) {
    console.log(err.message) // => "halted"
  }
  ```
  
 Can you move `await task.halt()` outside of the try catch?

Reply:
Ok, update to include that language. It's a good idea to demonstrate that `task.halt()` is not an error by moving it out of the `try/catch`

Reply:
Oh yeah, that's much better. What do you think about removing the comment and showing it in code.

```
import { run, sleep } from 'effection';

async function runExample() {
  // this task takes a long time to return
  let task = run(function*() {
    yield* sleep(10_000_000);
    return "hello world";
  });

  await task.halt();

  try {
    const value = await task // 💥 throws "halted" error;

    // will never reach here because halted task will never produce a value
    console.log(value);
  } catch(err) {
    console.log(err.message) // => "halted"
  }
}
```

Reply:
I like improving the flow of the code, but I don't think it's something we should leave unsaid. Repeating the same concept three times is a good thing, especially when it is very subtle.

I've updated the code to your example, but left the commentary in.